crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -2286,4 +2427,81 @@ pub(crate) mod tests {
         // The latch should be set now
         assert!(bob_identity.was_previously_verified());
     }
+
+    mod update_sender_data {
+        use std::collections::HashSet;
+
+        use assert_matches::assert_matches;
+        use matrix_sdk_test::async_test;
+        use ruma::room_id;
+
+        use super::{device_id, manager_test_helper};
+        use crate::{
+            identities::manager::testing::{other_user_id, user_id},
+            olm::{InboundGroupSession, SenderData},
+            store::{Changes, DeviceChanges},
+            Account, DeviceData, EncryptionSettings,
+        };
+
+        #[async_test]
+        async fn test_adds_device_info_to_existing_sessions() {
+            let manager = manager_test_helper(user_id(), device_id()).await;
+
+            // Given that we have lots of sessions in the store, from each of two devices
+            let account1 = Account::new(user_id());
+            let account2 = Account::new(other_user_id());
+
+            let mut account1_sessions = Vec::new();
+            for _ in 0..60 {
+                account1_sessions.push(create_inbound_group_session(&account1).await);
+            }
+            let mut account2_sessions = Vec::new();
+            for _ in 0..60 {
+                account2_sessions.push(create_inbound_group_session(&account2).await);
+            }
+            manager
+                .store
+                .save_changes(Changes {
+                    inbound_group_sessions: [account1_sessions.clone(), account2_sessions.clone()]
+                        .concat(),
+                    ..Default::default()
+                })
+                .await
+                .unwrap();
+
+            // When we get an update for one device
+            let device_data = DeviceData::from_account(&account1);
+            let updated_sessions = manager
+                .update_sender_data_from_device_changes(&DeviceChanges {
+                    changed: vec![device_data],
+                    ..Default::default()
+                })
+                .await
+                .unwrap();
+
+            // Then those sessions should be updated
+            assert_eq!(
+                updated_sessions.iter().map(|s| s.session_id()).collect::<HashSet<_>>(),
```

Comment:
A little helper function called something like "session_ids" would make this much easier for me to read, if you can be bothered.

Reply:
removed as a side-effect of 74e63219d